src/main/java/com/google/sps/servlets/NaturalLanguageServlet.java

```diff
@@ -63,9 +100,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         if (this.nlp == null) {
             this.nlp = new NaturalLanguageProcessor();
         }
+        if (this.db == null) {
+            this.db = new CaptionStorage();
+        }
 
         // Builds the Java object from JSON and preprocesses the captions by redefining time ranges
         YoutubeCaptions youtubeCaptions = gson.fromJson(json, YoutubeCaptions.class);
+
+        // Parses the video ID from the url
+        boolean addToDatabase = true;
+        String videoID = "";
+        String[] urlParts = youtubeCaptions.getVideoURL().split(VIDEO_URL_ID_DELIMITER);
+        if (urlParts.length <= 1) {
+            addToDatabase = false;
+        }
+        if (addToDatabase) {
+            videoID = urlParts[1];
+            int ampersandPosition = videoID.indexOf(URL_QUERY_DELIMITER);
+            if (ampersandPosition != -1) {
+                videoID = videoID.substring(0, ampersandPosition);
+            }
+            try {
+                db.addVideo(videoID, DB_NO_METADATA);
+            } catch (CaptionStorageException e) {
+                // TODO: report problem in metadataList at the end of this function
+            }
+        }
+
         int numCaptions = youtubeCaptions.getCaptions().size();
         NaturalLanguagePreprocessor preprocessor = new NaturalLanguagePreprocessor();
```

Comment:
do we need to create a new object, or we could use this.nlp?

Reply:
The `nlp` instance variable in this class refers to a NaturalLanguageProcessor object, not a NaturalLanguagePreprocessor object